### PR TITLE
[CI]: Add pull request verification

### PR DESCRIPTION
**Pull request by marchermans**
This might need some fiddling but this is how I envisioned TeamCity could do the pull request verification, it will run the `checkXXXX` tasks in a separate build pipeline.

If we want more detailed information we could technically split it apart into individual gradle invocations but this might cause additional delays due to repeated building and verification of some sub logic.

This PR targets the 1.18 branch as this is the current TeamCity main branch from which the default configuration and project layout are loaded.

### CODE REVIEW

**Comment by marchermans** [comment]
@SizableShrimp Is this what you are looking for?

File: .teamcity/settings.kts [hunk 1]
```diff
@@ -134,3 +135,31 @@ object PullRequests : BuildType({
...
+    id("MinecraftForge_MinecraftForge__PullRequestVerification")
+    name = "Pull Request Verification"
+    description = "Verifies pull requests for the project"
+
+    params {
+        checkbox("should_execute_build", "true", label = "Should build", description = "Indicates if the build task should be executed.", display = ParameterDisplay.HIDDEN,
+                checked = "true", unchecked = "false")
+        text(
+                "gradle_build_task",
+                "checkATs checkExcs checkSAS findFieldInstanceChecks checkLicenses checkPatches checkJarCompatibility",
```
Comment (sciwhiz12) [inline .teamcity/settings.kts:150]: Ideally, the CI should only need to invoke the `check` task (possibly excluding any `test` tasks) and we should only need configure the buildscript such that our existing `check*` tasks are ran by `check`.

**Review by sciwhiz12 — COMMENTED** [review]
Assuming this works as intended, it looks to be fine. 

However, we need to lay a bit of groundwork before this can be merged; a good amount of our check tasks currently aren't pure checks, but modify the data they're checking to correct them.

Obviously, we can't have the CI silently correcting errors (and subsequently discarding those corrections) in what are supposed to be verification tasks for pull requests. We have to refactor the check tasks such that they do _not_ modify the data, but only report failures so the CI can properly fail and alert the PR author.

**Comment by sciwhiz12** [comment]
Superseded by #9160.

**Pull request closed without merging**